pkg/occlient/occlient.go

```diff
@@ -782,12 +807,22 @@ func addBootstrapVolumeMount(dc *appsv1.DeploymentConfig, dcName string) {
 	}
 }
 
+// addEnvVar adds the environmental variables to the containers in the DC
+// dc is the deployment config to be updated
+// inputEnvVars is the array containing the corev1.EnvVar values
+func addEnvVar(dc *appsv1.DeploymentConfig, inputEnvVars []corev1.EnvVar) {
+	for i := range dc.Spec.Template.Spec.Containers {
+		dc.Spec.Template.Spec.Containers[i].Env = inputEnvVars
```

Comment:
> Also I see from https://github.com/redhat-developer/odo/pull/626/files#diff-6025e9b620e9321c03c11bf29e5c4dbdL517, that the dc created is always with only 1 container why so?

Simply because ODO currently doesn't support pods with multiple containers. There was no need for multi-container pods in use cases that ODO is trying to solve so far. So there is no way for users to create those.

Reply:
Most of our code simply expect that there is only one container in the DC. But here you are setting env variables for all containers. I think that we should do the same thing as in other places - error out if there is more than one container.  It will make sure that we don't have any unexpected behaviors if we start implementing multi-container solutions in the future (it is better to error out than do something not expected)